src/precice/impl/Participant.cpp

```diff
@@ -189,54 +178,80 @@ utils::ptr_vector<DataContext> &Participant::readDataContexts()
   return _readDataContexts;
 }
 
-bool Participant::isMeshUsed(
-    int meshID) const
+bool Participant::hasData(int dataID) const
 {
-  PRECICE_ASSERT((meshID >= 0) && (meshID < (int) _meshContexts.size()));
-  return _meshContexts[meshID] != nullptr;
+  return std::any_of(
+      _meshContexts.begin(), _meshContexts.end(),
+      [dataID](const auto mcptr) {
+        if (!mcptr) {
+          return false;
+        }
+        const auto &meshData = mcptr->mesh->data();
+        return std::any_of(meshData.begin(), meshData.end(), [dataID](const auto &dptr) {
+          return dptr->getID() == dataID;
+        });
+      });
 }
 
-bool Participant::isMeshProvided(
-    int meshID) const
+bool Participant::isDataUsed(const std::string &dataName, int meshID) const
 {
-  PRECICE_ASSERT((meshID >= 0) && (meshID < (int) _meshContexts.size()));
-  auto context = _meshContexts[meshID];
-  return (context != nullptr) && context->provideMesh;
+  const auto &meshData = meshContext(meshID).mesh->data();
+  const auto  match    = std::find_if(meshData.begin(), meshData.end(), [&dataName](auto &dptr) { return dptr->getName() == dataName; });
+  return match != meshData.end();
 }
 
-bool Participant::isDataUsed(
-    int dataID) const
+bool Participant::isDataUsed(int dataID) const
 {
   PRECICE_ASSERT((dataID >= 0) && (dataID < (int) _dataContexts.size()), dataID, (int) _dataContexts.size());
   return _dataContexts[dataID] != nullptr;
 }
 
-bool Participant::isDataRead(
-    int dataID) const
+bool Participant::isDataRead(int dataID) const
 {
   return std::any_of(_readDataContexts.begin(), _readDataContexts.end(), [dataID](const DataContext &context) {
     return context.toData->getID() == dataID;
   });
 }
 
-bool Participant::isDataWrite(
-    int dataID) const
+bool Participant::isDataWrite(int dataID) const
 {
   return std::any_of(_writeDataContexts.begin(), _writeDataContexts.end(), [dataID](const DataContext &context) {
     return context.fromData->getID() == dataID;
   });
 }
 
-const MeshContext &Participant::meshContext(
-    int meshID) const
+int Participant::getUsedDataID(const std::string &dataName, int meshID) const
+{
+  const auto &dptr = usedMeshContext(meshID).mesh->data(dataName);
+  PRECICE_ASSERT(dptr != nullptr);
+  return dptr->getID();
+}
+
+std::string Participant::getDataName(int dataID) const
+{
+  for (const MeshContext *mcptr : _meshContexts) {
+    if (mcptr) {
+      continue;
+    }
+    for (const auto &dptr : mcptr->mesh->data()) {
+      if (dptr->getID() == dataID) {
+        return dptr->getName();
+      }
+    }
+  }
+  PRECICE_UNREACHABLE("The dataID is invalid." << dataID);
```

Comment:
The `UNREACHABLE` macro uses `fmtlib`. Not completely sure here, but shouldn't you use the brackets `{}` in order to put the argument in the stream?

Reply:
This branch was still based on the non-fmtlib version of the macro.